asllib/doc/Statements.tex

```diff
@@ -3071,28 +3071,100 @@ \subsubsection{SemanticsRule.WriteFolder\label{sec:SemanticsRule.WriteFolder}}
+\Nstmt \derives \ & \Tprint \parsesep \Plist{\Nexpr} \parsesep \Tsemicolon & \\
+\Nstmt \derives \ & \Tprintln \parsesep \Plist{\Nexpr} \parsesep \Tsemicolon & \\
 \end{flalign*}
 
 \subsection{Abstract Syntax}
 \begin{flalign*}
-\stmt \derives\ & \SPrint(\overtext{\expr^*}{args}, \overtext{\Bool}{debug}) &
+  \stmt \derives\ & \SPrint(\overtext{\expr^*}{args}, \overtext{\Bool}{newline}) &
 \end{flalign*}
 
 \subsubsection{ASTRule.SPrint}
 \begin{mathpar}
-\inferrule{
-  \buildplist[\Nexpr](\vargs) \astarrow \astversion{\vargs}
-}{
+\inferrule{%
+  \buildplist[\Nexpr](\vargs) \astarrow \astversion{\vargs} \\
+  \vnewline \eqdef \False \\
+}{%
   \buildstmt(\overname{\Nstmt(\Tprint, \namednode{\vargs}{\Plist{\Nexpr}}, \Tsemicolon)}{\vparsednode})
   \astarrow
-  \overname{\SPrint(\astversion{\vargs})}{\vastnode}
+  \overname{\SPrint(\astversion{\vargs}, \vnewline)}{\vastnode}
+}
+\\
+\inferrule{%
+  \buildplist[\Nexpr](\vargs) \astarrow \astversion{\vargs} \\
+  \vnewline \eqdef \True \\
+  \vdebug \eqdef \False \\
+}{%
+  \buildstmt(\overname{\Nstmt(\Tprintln, \namednode{\vargs}{\Plist{\Nexpr}}, \Tsemicolon)}{\vparsednode})
+  \astarrow
+  \overname{\SPrint(\astversion{\vargs}, \vnewline)}{\vastnode}
 }
 \end{mathpar}
 
 \subsection{Typing}
+
+\subsubsection{TypingRule.SPrint\label{sec:TypingRule.SPrint}}
+
+\paragraph{Prose}
+All of the following apply:
+\begin{itemize}
+  \item $\vs$ denotes the print statement with arguments $\vargs$ and newline indicator $\vnewline$;
+  \item annotating each expression $\vargs_i$ in $\tenv$ yields $(\vt_i, \vargsp_i)$\ProseOrTypeError, for $i=1..k$;
+  \item checking that $\vt_i$ is a singular type yields $\True$\ProseOrTypeError, for each $i=1..k$;
+  \item $\news$ denotes the print statement with arguments $\vargsp$ and newline indicator $\vnewline$;
+  \item $\newtenv$ is $\tenv$.
+\end{itemize}
+
+\paragraph{Formally}
+\begin{mathpar}
+  \inferrule{%
+    \vi\in\listrange(\vargs): \annotateexpr{\vargs[i], \tenv} \typearrow (\vt_i, \vargsp[i])\OrTypeError \\
+    \vi\in\listrange(\vargs): \checktrans{\issingular(\vt_i)}{\BadPrintType} \typearrow \True \OrTypeError \\
+  }{%
+    \annotatestmt (\tenv, \overname{\SPrint(\vargs, \vnewline)}{\vs}) \typearrow (\SPrint(\vargsp, \vnewline), \tenv)
+  }
+\end{mathpar}
+
+\CodeSubsection{\SPrintBegin}{\SPrintEnd}{../Typing.ml}
+
 \subsection{Semantics}
 
+\subsubsection{Prose}
+All of the following apply:
+\begin{itemize}
+  \item $\vs$ denotes a Print statement with arguments $\elist$ and newline indicator $\vnewline$;
+  \item the evaluation of $\elist$ in $\env$ is $\Normal((\vlist, \vg), \newenv)$\ProseOrAbnormal;
+  \item \hyperlink{print-literals}{prints} all the elements in $\elist$, without separator;
+  \item if $\vnewline$ is $\True$, prints a newline character;
+  \item $\newenv$ is $\env$.
+\end{itemize}
+
+\subsubsection{Formally}
+
+Since it is not mandatory for an ASL backend to have a console component, this
```

Comment:
I suggest removing this text and replacing it with the one at the beginning of the semantics subsection.